GitCommands/Git/Executable.cs

```diff
@@ -111,7 +111,15 @@ private sealed class ProcessWrapper : IProcess
                 try
                 {
                     _process.Start();
-                    _logOperation.SetProcessId(_process.Id);
+                    try
+                    {
+                        _logOperation.SetProcessId(_process.Id);
+                    }
+                    catch (InvalidOperationException ex) when (useShellExecute)
+                    {
+                        // _process.Start() has succeeded, ignore the failure getting the _process.Id
+                        _logOperation.LogProcessEnd(ex);
```

Comment:
I assume that the actual exit will still update command and results?

Reply:
> I assume that the actual exit will still update command and results?

The call is visible and - to my surprise - even with a duration (see screenshot in the PR description).